go/vt/vtgate/evalengine/eval.go

```diff
@@ -226,7 +229,16 @@ func valueToEvalCast(v sqltypes.Value, typ sqltypes.Type) (eval, error) {
 			fval, _ := fastparse.ParseFloat64(v.RawStr())
 			return newEvalFloat(fval), nil
 		default:
-			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "coercion should not try to coerce this value to a float: %v", v)
+			e, err := valueToEval(v, collations.TypedCollation{
```

Comment:
we are using this snippet in a few places. worth extracting into a method?

```go
collations.TypedCollation{
  Collation:    collation,
  Coercibility: collations.CoerceCoercible,
  Repertoire:   collations.RepertoireUnicode,
}
```

Reply:
@systay that method even exists already! But I'm not using it, fixing it up.